invokeai/app/invocations/controlnet_image_processors.py

```diff
@@ -559,3 +559,32 @@ def show_anns(self, anns: List[Dict]):
             img[:, :] = ann_color
             final_img.paste(Image.fromarray(img, mode="RGB"), (0, 0), Image.fromarray(np.uint8(m * 255)))
         return np.array(final_img, dtype=np.uint8)
+
+
+@invocation(
+    "color_map_image_processor",
+    title="Color Map Processor",
+    tags=["controlnet"],
+    category="controlnet",
+    version="1.0.0",
+)
+class ColorMapImageProcessorInvocation(ImageProcessorInvocation):
+    """Generates a color map from the provided image"""
+
+    map_resolution: int = InputField(default=512, ge=0, description=FieldDescriptions.image_res)
```

Comment:
I don't think it makes sense for the `ColorMapImageProcessorInvocation` to be responsible for resizing to a target `map_resolution`. None of the other image processor invocations do this. It feels like it would be better-handled by a dedicated resizing node with support for various cropping/rescaling modes (e.g. like the modes in https://github.com/invoke-ai/InvokeAI/blob/b152fbf72f163e4a60bda7a54747d0a752d63ca1/invokeai/app/util/controlnet_utils.py#L267).

(_If_ we do keep this parameter, we should expose both the width and height. It's not necessary to constrain to a square.)

Reply:
I agree. Included in this because I realized we don't really use it anywhere else but I do think we should. There is minimal difference between a large res control image vs a smaller one from my testing.